.github/workflows/ci.yaml

```diff
@@ -20,28 +20,119 @@ on:
 jobs:
   lint:
     name: Lint
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v5.1.2
+
+  build:
+    name: Build charm
+    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v5.1.2
+    permissions:
+      actions: write  # Needed to manage GitHub Actions cache
+
+  gh-hosted-integration-k8s-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        tox-environment:
+          - integration-k8s-smoke
+        juju-snap-channel:  ["2.9/stable", "3.1/stable"]
+        include:
+          - juju-snap-channel: "3.1/stable"
+            agent-version: "3.1.6"
+            libjuju-version:  "3.2.2"
+          - juju-snap-channel: "2.9/stable"
+            agent-version: "2.9.45"
+            libjuju-version:  "2.9.45.0"
+    name: ${{ matrix.juju-snap-channel }} - (GH hosted) ${{ matrix.tox-environment }}
+    needs:
+      - lint
+      - build
     runs-on: ubuntu-latest
-    timeout-minutes: 5
+    timeout-minutes: 120
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
-      - name: Install tox
-        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
-        run: python3 -m pip install tox
-      - name: Run linters
-        run: tox run -e lint
+        uses: actions/checkout@v3
```

Comment:
Out of curiosity: is there any reason to downgrade from v4 to v3? (i.e. compatibility with the workflow)

Reply:
The reason is copypaste. Will bump back.